src/Example/UserRegistrationBundle/Controller/UserRegistrationController.php

```diff
@@ -0,0 +1,196 @@
+use Example\UserRegistrationBundle\Form\Type\UserRegistrationType;
+
+/**
+ * @Route("/users/registration", service="example_user_registration.user_registration_controller")
+ * @Pageflow({
+ *     @StartPage({"input",
+ *         @Transition("confirmation"),
+ *     }),
+ *     @Page({"confirmation",
+ *         @Transition("success"),
+ *         @Transition("input")
+ *     }),
+ *     @EndPage("success")
+ * })
+ */
+class UserRegistrationController extends Controller implements ConversationalControllerInterface
+{
+    const VIEW_INPUT = 'ExampleUserRegistrationBundle:UserRegistration:input.html.twig';
+    const VIEW_CONFIRMATION = 'ExampleUserRegistrationBundle:UserRegistration:confirmation.html.twig';
+    const VIEW_SUCCESS = 'ExampleUserRegistrationBundle:UserRegistration:success.html.twig';
+
+    /**
+     * {@inheritDoc}
+     */
+    private $conversationContext;
+
+    /**
+     * @var User
+     *
+     * @Stateful
+     */
+    private $user;
+
+    /**
+     * {@inheritDoc}
+     */
+    public function setConversationContext(ConversationContext $conversationContext)
+    {
+        $this->conversationContext = $conversationContext;
+    }
+
+    /**
+     * @Init
+     */
+    public function initialize()
+    {
+        $this->user = new User();
+    }
+
+    /**
+     * @return Response
+     *
+     * @Route("/")
+     * @Method("GET")
+     * @Accept("input")
+     * @Accept("confirmation")
+     */
+    public function inputGetAction()
+    {
+        if ($this->conversationContext->getConversation()->getCurrentPage()->getPageId() == 'confirmation') {
+            $this->conversationContext->getConversation()->transition('input');
+        }
+
+        $form = $this->createForm(new UserRegistrationType(), $this->user, array('action' => $this->generateUrl('example_userregistration_userregistration_inputpost'), 'method' => 'POST'));
+
+        return $this->render(self::VIEW_INPUT, array(
+            'form' => $form->createView(),
+        ));
+    }
+
+    /**
+     * @param  Request  $request
+     * @return Response
+     *
+     * @Route("/")
+     * @Method("POST")
+     * @Accept("input")
+     * @Accept("confirmation")
+     */
+    public function inputPostAction(Request $request)
+    {
+        if ($this->conversationContext->getConversation()->getCurrentPage()->getPageId() == 'confirmation') {
+            $this->conversationContext->getConversation()->transition('input');
+        }
+
+        $form = $this->createForm(new UserRegistrationType(), $this->user, array('action' => $this->generateUrl('example_userregistration_userregistration_inputpost'), 'method' => 'POST'));
+        $form->handleRequest($request);
+
+        if ($form->isValid()) {
+            $this->conversationContext->getConversation()->transition('confirmation');
+
+            return $this->redirect($this->conversationContext->generateUrl('example_userregistration_userregistration_confirmationget'));
+        } else {
```

Comment:
__[development][phpmd]__
* The method inputPostAction uses an else expression. Else is never necessary and you can simplify the code to work without else.